test/gov/hzn_sdo.sh

```diff
@@ -0,0 +1,1166 @@
+		exit 1
+	fi
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo voucher download': exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+
+echo -e "${PREFIX} Testing 'hzn sdo voucher download <device-id> -f /tmp/test_voucher.json' when /tmp/test_voucher.json already exists"
+cmdOutput=$(hzn sdo voucher download $VOUCHER_DEVICE_ID -f /tmp/test_voucher.json 2>&1)
+rc=$?
+if [[ $rc -eq 1 && "$cmdOutput" == *'Error:'*'File'*'already exists. Please specify a different file path or file name.'* ]]; then
+	echo -e "${PREFIX} received expected error response."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo voucher download' when /tmp/test_voucher.json already exists: exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+
+echo -e "${PREFIX} Testing 'hzn sdo voucher download <device-id> -f /tmp/test_voucher.json -O' when /tmp/test_voucher.json already exists"
+cmdOutput=$(hzn sdo voucher download $VOUCHER_DEVICE_ID -f /tmp/test_voucher.json -O 2>&1)
+rc=$?
+if [[ $rc -eq 0 ]]; then
+	if [[ -f /tmp/test_voucher.json && "$(cat /tmp/test_voucher.json)" == "$(cat /tmp/sdo_voucher.json)" ]]; then
+		echo -e "${PREFIX} completed."
+	elif [[ ! -f /tmp/test_voucher.json ]]; then
+		echo -e "${PREFIX} Failed: 'hzn sdo voucher download <device-id> -f /tmp/test_voucher.json' not downloaded to /tmp/test_voucher.json"
+		exit 1
+	else
+		echo -e "${PREFIX} Failed: 'hzn sdo voucher download <device-id> -f /tmp/test_voucher.json': downloaded file does not match expected output."
+		exit 1
+	fi
+elif [[ $rc -eq 1 && "$cmdOutput" == *'Error:'*'File'*'already exists. Please specify a different file path or file name.'* ]]; then
+	echo -e "${PREFIX} Failed: 'hzn sdo voucher download' did not overwrite output file with -O flag set."
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo voucher download': exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+rm /tmp/test_voucher.json
+
+echo -e "${PREFIX} Testing 'hzn sdo voucher download <device-id> -f /tmp'"
+cmdOutput=$(hzn sdo voucher download $VOUCHER_DEVICE_ID -f /tmp 2>&1)
+rc=$?
+if [[ $rc -eq 0 ]]; then
+	if [[ -f /tmp/$VOUCHER_DEVICE_ID.json && "$(cat /tmp/$VOUCHER_DEVICE_ID.json)" == "$(cat /tmp/sdo_voucher.json)" ]]; then
+		echo -e "${PREFIX} completed."
+	elif [[ ! -f /tmp/$VOUCHER_DEVICE_ID.json ]]; then
+		echo -e "${PREFIX} Failed: 'hzn sdo voucher download <device-id> -f /tmp' not downloaded to /tmp/$VOUCHER_DEVICE_ID.json"
+		exit 1
+	else
+		echo -e "${PREFIX} Failed: 'hzn sdo voucher download <device-id> -f /tmp': downloaded file does not match expected output."
+		exit 1
+	fi
+else
+	echo -e "${PREFIX} Failed: Wrong error response from 'hzn sdo voucher download': exit code: $rc, output: $cmdOutput."
+	exit 1
+fi
+rm /tmp/$VOUCHER_DEVICE_ID.json
+
+# echo -e "${PREFIX} Testing 'hzn sdo voucher download' with incorrect voucher name argument"
```

Comment:
Why is this test case commented out?

Reply:
I did that because the API updates were not reflected in the test environment which would cause these tests to fail. They have now been updated and uncommented.